### PR TITLE
Increase file count after sharing

### PR DESCRIPTION
**Pull request by luka-nextcloud**
Signed-off-by: Luka Trovic <luka@nextcloud.com>

* Resolves: #3677
* Target version: master 

### Summary
File count does not increase after sharing.

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [ ] Tests (unit, integration, api and/or acceptance) are included
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: src/components/card/AttachmentList.vue [hunk 1]
```diff
@@ -234,6 +234,7 @@ export default {
 						shareWith: '' + this.cardId,
 					}).then(() => {
 						this.$store.dispatch('fetchAttachments', this.cardId)
+						this.$store.commit('cardIncreaseAttachmentCount', this.cardId)
```
Comment (vinicius73) [inline src/components/card/AttachmentList.vue:237]: Maybe updating the value direct into `fetchAttachments`  action will be more consistent.

File: lib/Service/AttachmentService.php [hunk 2]
```diff
@@ -137,16 +137,17 @@ public function findAll($cardId, $withDeleted = false) {
...
 	 */
-	public function count($cardId) {
+	public function count($cardId, $update = false) {
 		if (is_numeric($cardId) === false) {
 			throw new BadRequestException('card id must be a number');
 		}
 
 		$count = $this->cache->get('card-' . $cardId);
-		if (!$count) {
+		if ($update OR !$count) {
```
Comment (juliushaertl) [inline lib/Service/AttachmentService.php:150]:
```suggestion
		if ($update || !$count) {
```

Just to stick to our code style ;)

File: lib/Sharing/DeckShareProvider.php [hunk 3]
```diff
@@ -75,16 +76,29 @@ class DeckShareProvider implements \OCP\Share\IShareProvider {
...
+		CardMapper $cardMapper, 
+		AttachmentService $attachmentService,
+		PermissionService $permissionService, 
+		IL10N $l
+	) {
 		$this->dbConnection = $connection;
 		$this->shareManager = $shareManager;
 		$this->boardMapper = $boardMapper;
 		$this->cardMapper = $cardMapper;
+		$this->attachmentService = $attachmentService;
```
Comment (juliushaertl) [inline lib/Sharing/DeckShareProvider.php:99]: Injecting this service seems to cause a cyclic dependency and therefore crashing the php process on the test runs

**Comment by luka-nextcloud** [comment]
@juliushaertl Do we need to add a test for it now?

**Comment by juliushaertl** [comment]
If existing tests pass that should be fine for the backend side I'd say with the integration tests in https://github.com/nextcloud/deck/blob/master/tests/integration/features/sharing.feature

For the frontend I've started collecting the case in https://github.com/nextcloud/deck/issues/3768 so we can add it after the cypress parts are merged.

**Comment by juliushaertl** [comment]
Seems tests are not happy yet unfortunately ;)

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**